src/backend/opengl/chart_impl.cpp

```diff
@@ -353,6 +353,18 @@ void chart2d_impl::generateChartData()
     CheckGL("End chart2d_impl::generateChartData");
 }
 
+int getDigitCount(int value)
```

Comment:
I don't think this would handle the case of min and max being something like 0.000001, which would also not be rendered correctly.

I believe what you should do is accept an of double as a param, `abs` it, and then check whether the number is greater than 1 or not.
If it is greater than one, convert it to int and then do what you are doing.
If it is between 0 and 1, then multiply by 10 until the number is greater than 1. That will give you significant digits right of the decimal.